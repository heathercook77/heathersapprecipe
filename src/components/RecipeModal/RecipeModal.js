import React from "react";
import { Card, CloseButton, Modal } from "react-bootstrap";
import "./index.css";

export default function RecipeModal(props) {
  return (
    <>
      {props.show && (
        <Modal show={true} onHide={props.handleClose}>
          <Modal.Header closeButton>
            <Modal.Title>Recipe</Modal.Title>
          </Modal.Header>
          <Modal.Body>
            <Card style={{ width: "18rem" }}>
              <Card.Img
                variant="top"
                src={props.recipeData.image}
                className="recipe-img-info"
              />
              <Card.Body>
                <Card.Title>{props.recipeData.label}</Card.Title>
                <h6 className="ing-heading">Ingredients-</h6>
                <ul className="ingredients-list">
                  {props.recipeData.ingredients.map((ing, i) => (
                    <li className="ing-li" key={i}>
                      {ing.text}
                      <img src={ing.image} alt="ing" className="ing-img" />
                    </li>
                  ))}
                </ul>
                <Card.Text>
                  Recipe is taken from {props.recipeData.source}.<br />
                  You may find the complete recipe at the following link:
                  <br />
                  <a
                    href={props.recipeData.url}
                    target="_blank"
                    rel="noreferrer"
                    className="link-to-source"
                  >
                    {props.recipeData.source}
                  </a>
                </Card.Text>
              </Card.Body>
            </Card>
          </Modal.Body>
          <Modal.Footer>
            <CloseButton
              className="close-modal-lower-btn"
              variant="secondary"
              onClick={props.handleClose}
            >
              Close
            </CloseButton>
          </Modal.Footer>
        </Modal>
      )}
    </>
  );
}
