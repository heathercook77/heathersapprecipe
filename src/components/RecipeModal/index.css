.btn-close::after {
  content: "X";
}
.close-modal-lower-btn::after {
  content: "";
}
.card {
  width: 100% !important;
}
.recipe-img-info {
  height: 250px;
  object-fit: scale-down;
}
.ing-heading {
}
.ingredients-list {
}
.ing-li {
  font-size: 14px;
  color: rgb(126, 126, 126);
}
.ing-img {
  border-radius: 4px;
  margin-left: 7px;
  width: 30px;
  height: 30px;
}
