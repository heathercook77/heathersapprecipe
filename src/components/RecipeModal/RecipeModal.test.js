import RecipeModal from "./RecipeModal";
import React from "react";
import { shallow } from "enzyme";
import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });
describe("Modal Component", () => {
  let wrapper, props;
  props = {
    handleClose: jest.fn(),
    show: true,
    recipeData: {
      uri: "http://www.edamam.com/ontologies/edamam.owl#recipe_0ff11a2807e1d4d99bdb33de5d2fd9bd",
      label: "Creamsicle Pudding Recipe",
      image:
        "https://www.edamam.com/web-img/ac7/ac760a4ed728d98568d7db8e6efd6489.jpg",
      source: "Food52",
      url: "https://food52.com/recipes/9752-creamsicle-pudding-recipe",
      shareAs:
        "http://www.edamam.com/recipe/creamsicle-pudding-recipe-0ff11a2807e1d4d99bdb33de5d2fd9bd/pudding",
      yield: 2,
      dietLabels: ["Balanced", "Low-Sodium"],
      healthLabels: [
        "Vegetarian",
        "Pescatarian",
        "Gluten-Free",
        "Wheat-Free",
        "Peanut-Free",
        "Tree-Nut-Free",
        "Soy-Free",
        "Fish-Free",
        "Shellfish-Free",
        "Pork-Free",
        "Red-Meat-Free",
        "Crustacean-Free",
        "Celery-Free",
        "Mustard-Free",
        "Sesame-Free",
        "Lupine-Free",
        "Mollusk-Free",
        "Alcohol-Free",
        "Kosher",
      ],
      cautions: ["Gluten", "Wheat", "Sulfites"],
      ingredientLines: [
        "2 tablespoons sugar",
        "1 tablespoon cornstarch",
        "1 egg yolk, slightly beaten in a bowl and set aside",
        "1 cup cup milk (for vanilla pudding)",
        "1 1/2 teaspoon butter",
        "1 vanilla bean, split with seeds (for vanilla pudding)",
        "1/4 cup oj concentrate (for orange pudding)",
        "3/4 cup milk (for orange pudding)",
      ],
      ingredients: [
        {
          text: "2 tablespoons sugar",
          quantity: 2,
          measure: "tablespoon",
          food: "sugar",
          weight: 24.9999999995774,
          foodCategory: "sugars",
          image:
            "https://www.edamam.com/food-img/ecb/ecb3f5aaed96d0188c21b8369be07765.jpg",
        },
        {
          text: "1 tablespoon cornstarch",
          quantity: 1,
          measure: "tablespoon",
          food: "cornstarch",
          weight: 7.999999999864745,
          foodCategory: null,
          image:
            "https://www.edamam.com/food-img/f9b/f9b74d9495b40c0aea955c37a1fc39dc.jpg",
        },
        {
          text: "1 egg yolk, slightly beaten in a bowl and set aside",
          quantity: 1,
          measure: "<unit>",
          food: "egg yolk",
          weight: 13.466666666666667,
          foodCategory: "Eggs",
          image:
            "https://www.edamam.com/food-img/e3f/e3f697887aec0e8ecf09a9ca8ef3944a.jpg",
        },
        {
          text: "1 cup cup milk (for vanilla pudding)",
          quantity: 1,
          measure: "cup",
          food: "milk",
          weight: 244,
          foodCategory: "Milk",
          image:
            "https://www.edamam.com/food-img/7c9/7c9962acf83654a8d98ea6a2ade93735.jpg",
        },
        {
          text: "1 1/2 teaspoon butter",
          quantity: 1.5,
          measure: "teaspoon",
          food: "butter",
          weight: 7.093750000359803,
          foodCategory: "Dairy",
          image:
            "https://www.edamam.com/food-img/713/71397239b670d88c04faa8d05035cab4.jpg",
        },
        {
          text: "1 vanilla bean, split with seeds (for vanilla pudding)",
          quantity: 1,
          measure: "<unit>",
          food: "vanilla bean",
          weight: 5,
          foodCategory: "Condiments and sauces",
          image:
            "https://www.edamam.com/food-img/90f/90f910b0bf82750d4f6528263e014cca.jpg",
        },
        {
          text: "1/4 cup oj concentrate (for orange pudding)",
          quantity: 0.25,
          measure: "cup",
          food: "orange",
          weight: 45,
          foodCategory: "fruit",
          image:
            "https://www.edamam.com/food-img/8ea/8ea264a802d6e643c1a340a77863c6ef.jpg",
        },
        {
          text: "3/4 cup milk (for orange pudding)",
          quantity: 0.75,
          measure: "cup",
          food: "milk",
          weight: 183,
          foodCategory: "Milk",
          image:
            "https://www.edamam.com/food-img/7c9/7c9962acf83654a8d98ea6a2ade93735.jpg",
        },
      ],
      calories: 517.4748541670956,
      totalWeight: 530.5604166664687,
      totalTime: 512,
      cuisineType: ["british"],
      mealType: ["lunch/dinner"],
      dishType: ["desserts"],
      totalNutrients: {
        ENERC_KCAL: {
          label: "Energy",
          quantity: 517.4748541670956,
          unit: "kcal",
        },
        FAT: {
          label: "Total lipid (fat)",
          quantity: 23.2662939586251,
          unit: "g",
        },
        FASAT: {
          label: "Fatty acids, total saturated",
          quantity: 12.901638833518145,
          unit: "g",
        },
        FATRN: {
          label: "Fatty acids, total trans",
          quantity: 0.23253312501179432,
          unit: "g",
        },
        FAMS: {
          label: "Fatty acids, total monounsaturated",
          quantity: 6.551264520908946,
          unit: "g",
        },
        FAPU: {
          label: "Fatty acids, total polyunsaturated",
          quantity: 1.6281014791775816,
          unit: "g",
        },
        CHOCDF: {
          label: "Carbohydrate, by difference",
          quantity: 59.2003095827876,
          unit: "g",
        },
        FIBTG: {
          label: "Fiber, total dietary",
          quantity: 1.1519999999987829,
          unit: "g",
        },
        SUGAR: {
          label: "Sugars, total",
          quantity: 51.4331695829118,
          unit: "g",
        },
        "SUGAR.added": {
          label: "Sugars, added",
          quantity: 24.949999999578246,
          unit: "g",
        },
        PROCNT: {
          label: "Protein",
          quantity: 16.09341020833604,
          unit: "g",
        },
        CHOLE: {
          label: "Cholesterol",
          quantity: 204.06489583410695,
          unit: "mg",
        },
        NA: {
          label: "Sodium, Na",
          quantity: 192.2743125000232,
          unit: "mg",
        },
        CA: {
          label: "Calcium, Ca",
          quantity: 520.5445000000793,
          unit: "mg",
        },
        MG: {
          label: "Magnesium, Mg",
          quantity: 48.85520833333648,
          unit: "mg",
        },
        K: {
          label: "Potassium, K",
          quantity: 669.6111666667405,
          unit: "mg",
        },
        FE: {
          label: "Iron, Fe",
          quantity: 0.5982587499992249,
          unit: "mg",
        },
        ZN: {
          label: "Zinc, Zn",
          quantity: 1.940317708333534,
          unit: "mg",
        },
        P: {
          label: "Phosphorus, P",
          quantity: 420.5425000000688,
          unit: "mg",
        },
        VITA_RAE: {
          label: "Vitamin A, RAE",
          quantity: 301.19925000246104,
          unit: "µg",
        },
        VITC: {
          label: "Vitamin C, total ascorbic acid",
          quantity: 23.94,
          unit: "mg",
        },
        THIA: {
          label: "Thiamin",
          quantity: 0.26017602083335134,
          unit: "mg",
        },
        RIBF: {
          label: "Riboflavin",
          quantity: 0.822645875000042,
          unit: "mg",
        },
        NIA: {
          label: "Niacin",
          quantity: 0.5343913750001511,
          unit: "mg",
        },
        VITB6A: {
          label: "Vitamin B-6",
          quantity: 0.2293661458333441,
          unit: "mg",
        },
        FOLDFE: {
          label: "Folate, DFE",
          quantity: 54.724145833344124,
          unit: "µg",
        },
        FOLFD: {
          label: "Folate, food",
          quantity: 54.724145833344124,
          unit: "µg",
        },
        FOLAC: {
          label: "Folic acid",
          quantity: 0,
          unit: "µg",
        },
        VITB12: {
          label: "Vitamin B-12",
          quantity: 2.196159375000612,
          unit: "µg",
        },
        VITD: {
          label: "Vitamin D (D2 + D3)",
          quantity: 6.3846062500053975,
          unit: "µg",
        },
        TOCPHA: {
          label: "Vitamin E (alpha-tocopherol)",
          quantity: 0.8919150000083474,
          unit: "mg",
        },
        VITK1: {
          label: "Vitamin K (phylloquinone)",
          quantity: 1.8718291666918527,
          unit: "µg",
        },
        WATER: {
          label: "Water",
          quantity: 426.9692320833865,
          unit: "g",
        },
      },
      totalDaily: {
        ENERC_KCAL: {
          label: "Energy",
          quantity: 25.87374270835478,
          unit: "%",
        },
        FAT: {
          label: "Fat",
          quantity: 35.794298397884766,
          unit: "%",
        },
        FASAT: {
          label: "Saturated",
          quantity: 64.50819416759073,
          unit: "%",
        },
        CHOCDF: {
          label: "Carbs",
          quantity: 19.733436527595867,
          unit: "%",
        },
        FIBTG: {
          label: "Fiber",
          quantity: 4.6079999999951315,
          unit: "%",
        },
        PROCNT: {
          label: "Protein",
          quantity: 32.18682041667208,
          unit: "%",
        },
        CHOLE: {
          label: "Cholesterol",
          quantity: 68.0216319447023,
          unit: "%",
        },
        NA: {
          label: "Sodium",
          quantity: 8.011429687500966,
          unit: "%",
        },
        CA: {
          label: "Calcium",
          quantity: 52.05445000000793,
          unit: "%",
        },
        MG: {
          label: "Magnesium",
          quantity: 11.632192460318208,
          unit: "%",
        },
        K: {
          label: "Potassium",
          quantity: 14.24704609929235,
          unit: "%",
        },
        FE: {
          label: "Iron",
          quantity: 3.3236597222179163,
          unit: "%",
        },
        ZN: {
          label: "Zinc",
          quantity: 17.639251893941218,
          unit: "%",
        },
        P: {
          label: "Phosphorus",
          quantity: 60.077500000009834,
          unit: "%",
        },
        VITA_RAE: {
          label: "Vitamin A",
          quantity: 33.466583333606785,
          unit: "%",
        },
        VITC: {
          label: "Vitamin C",
          quantity: 26.6,
          unit: "%",
        },
        THIA: {
          label: "Thiamin (B1)",
          quantity: 21.681335069445947,
          unit: "%",
        },
        RIBF: {
          label: "Riboflavin (B2)",
          quantity: 63.28045192308016,
          unit: "%",
        },
        NIA: {
          label: "Niacin (B3)",
          quantity: 3.339946093750944,
          unit: "%",
        },
        VITB6A: {
          label: "Vitamin B6",
          quantity: 17.643549679488007,
          unit: "%",
        },
        FOLDFE: {
          label: "Folate equivalent (total)",
          quantity: 13.681036458336031,
          unit: "%",
        },
        VITB12: {
          label: "Vitamin B12",
          quantity: 91.50664062502551,
          unit: "%",
        },
        VITD: {
          label: "Vitamin D",
          quantity: 42.56404166670265,
          unit: "%",
        },
        TOCPHA: {
          label: "Vitamin E",
          quantity: 5.94610000005565,
          unit: "%",
        },
        VITK1: {
          label: "Vitamin K",
          quantity: 1.5598576389098773,
          unit: "%",
        },
      },
      digest: [
        {
          label: "Fat",
          tag: "FAT",
          schemaOrgTag: "fatContent",
          total: 23.2662939586251,
          hasRDI: true,
          daily: 35.794298397884766,
          unit: "g",
          sub: [
            {
              label: "Saturated",
              tag: "FASAT",
              schemaOrgTag: "saturatedFatContent",
              total: 12.901638833518145,
              hasRDI: true,
              daily: 64.50819416759073,
              unit: "g",
            },
            {
              label: "Trans",
              tag: "FATRN",
              schemaOrgTag: "transFatContent",
              total: 0.23253312501179432,
              hasRDI: false,
              daily: 0,
              unit: "g",
            },
            {
              label: "Monounsaturated",
              tag: "FAMS",
              schemaOrgTag: null,
              total: 6.551264520908946,
              hasRDI: false,
              daily: 0,
              unit: "g",
            },
            {
              label: "Polyunsaturated",
              tag: "FAPU",
              schemaOrgTag: null,
              total: 1.6281014791775816,
              hasRDI: false,
              daily: 0,
              unit: "g",
            },
          ],
        },
        {
          label: "Carbs",
          tag: "CHOCDF",
          schemaOrgTag: "carbohydrateContent",
          total: 59.2003095827876,
          hasRDI: true,
          daily: 19.733436527595867,
          unit: "g",
          sub: [
            {
              label: "Carbs (net)",
              tag: "CHOCDF.net",
              schemaOrgTag: null,
              total: 0,
              hasRDI: false,
              daily: 0,
              unit: "g",
            },
            {
              label: "Fiber",
              tag: "FIBTG",
              schemaOrgTag: "fiberContent",
              total: 1.1519999999987829,
              hasRDI: true,
              daily: 4.6079999999951315,
              unit: "g",
            },
            {
              label: "Sugars",
              tag: "SUGAR",
              schemaOrgTag: "sugarContent",
              total: 51.4331695829118,
              hasRDI: false,
              daily: 0,
              unit: "g",
            },
            {
              label: "Sugars, added",
              tag: "SUGAR.added",
              schemaOrgTag: null,
              total: 24.949999999578246,
              hasRDI: false,
              daily: 0,
              unit: "g",
            },
          ],
        },
        {
          label: "Protein",
          tag: "PROCNT",
          schemaOrgTag: "proteinContent",
          total: 16.09341020833604,
          hasRDI: true,
          daily: 32.18682041667208,
          unit: "g",
        },
        {
          label: "Cholesterol",
          tag: "CHOLE",
          schemaOrgTag: "cholesterolContent",
          total: 204.06489583410695,
          hasRDI: true,
          daily: 68.0216319447023,
          unit: "mg",
        },
        {
          label: "Sodium",
          tag: "NA",
          schemaOrgTag: "sodiumContent",
          total: 192.2743125000232,
          hasRDI: true,
          daily: 8.011429687500966,
          unit: "mg",
        },
        {
          label: "Calcium",
          tag: "CA",
          schemaOrgTag: null,
          total: 520.5445000000793,
          hasRDI: true,
          daily: 52.05445000000793,
          unit: "mg",
        },
        {
          label: "Magnesium",
          tag: "MG",
          schemaOrgTag: null,
          total: 48.85520833333648,
          hasRDI: true,
          daily: 11.632192460318208,
          unit: "mg",
        },
        {
          label: "Potassium",
          tag: "K",
          schemaOrgTag: null,
          total: 669.6111666667405,
          hasRDI: true,
          daily: 14.24704609929235,
          unit: "mg",
        },
        {
          label: "Iron",
          tag: "FE",
          schemaOrgTag: null,
          total: 0.5982587499992249,
          hasRDI: true,
          daily: 3.3236597222179163,
          unit: "mg",
        },
        {
          label: "Zinc",
          tag: "ZN",
          schemaOrgTag: null,
          total: 1.940317708333534,
          hasRDI: true,
          daily: 17.639251893941218,
          unit: "mg",
        },
        {
          label: "Phosphorus",
          tag: "P",
          schemaOrgTag: null,
          total: 420.5425000000688,
          hasRDI: true,
          daily: 60.077500000009834,
          unit: "mg",
        },
        {
          label: "Vitamin A",
          tag: "VITA_RAE",
          schemaOrgTag: null,
          total: 301.19925000246104,
          hasRDI: true,
          daily: 33.466583333606785,
          unit: "µg",
        },
        {
          label: "Vitamin C",
          tag: "VITC",
          schemaOrgTag: null,
          total: 23.94,
          hasRDI: true,
          daily: 26.6,
          unit: "mg",
        },
        {
          label: "Thiamin (B1)",
          tag: "THIA",
          schemaOrgTag: null,
          total: 0.26017602083335134,
          hasRDI: true,
          daily: 21.681335069445947,
          unit: "mg",
        },
        {
          label: "Riboflavin (B2)",
          tag: "RIBF",
          schemaOrgTag: null,
          total: 0.822645875000042,
          hasRDI: true,
          daily: 63.28045192308016,
          unit: "mg",
        },
        {
          label: "Niacin (B3)",
          tag: "NIA",
          schemaOrgTag: null,
          total: 0.5343913750001511,
          hasRDI: true,
          daily: 3.339946093750944,
          unit: "mg",
        },
        {
          label: "Vitamin B6",
          tag: "VITB6A",
          schemaOrgTag: null,
          total: 0.2293661458333441,
          hasRDI: true,
          daily: 17.643549679488007,
          unit: "mg",
        },
        {
          label: "Folate equivalent (total)",
          tag: "FOLDFE",
          schemaOrgTag: null,
          total: 54.724145833344124,
          hasRDI: true,
          daily: 13.681036458336031,
          unit: "µg",
        },
        {
          label: "Folate (food)",
          tag: "FOLFD",
          schemaOrgTag: null,
          total: 54.724145833344124,
          hasRDI: false,
          daily: 0,
          unit: "µg",
        },
        {
          label: "Folic acid",
          tag: "FOLAC",
          schemaOrgTag: null,
          total: 0,
          hasRDI: false,
          daily: 0,
          unit: "µg",
        },
        {
          label: "Vitamin B12",
          tag: "VITB12",
          schemaOrgTag: null,
          total: 2.196159375000612,
          hasRDI: true,
          daily: 91.50664062502551,
          unit: "µg",
        },
        {
          label: "Vitamin D",
          tag: "VITD",
          schemaOrgTag: null,
          total: 6.3846062500053975,
          hasRDI: true,
          daily: 42.56404166670265,
          unit: "µg",
        },
        {
          label: "Vitamin E",
          tag: "TOCPHA",
          schemaOrgTag: null,
          total: 0.8919150000083474,
          hasRDI: true,
          daily: 5.94610000005565,
          unit: "mg",
        },
        {
          label: "Vitamin K",
          tag: "VITK1",
          schemaOrgTag: null,
          total: 1.8718291666918527,
          hasRDI: true,
          daily: 1.5598576389098773,
          unit: "µg",
        },
        {
          label: "Sugar alcohols",
          tag: "Sugar.alcohol",
          schemaOrgTag: null,
          total: 0,
          hasRDI: false,
          daily: 0,
          unit: "g",
        },
        {
          label: "Water",
          tag: "WATER",
          schemaOrgTag: null,
          total: 426.9692320833865,
          hasRDI: false,
          daily: 0,
          unit: "g",
        },
      ],
    },
  };

  wrapper = shallow(<RecipeModal {...props} />);

  it("should render modal with data", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
