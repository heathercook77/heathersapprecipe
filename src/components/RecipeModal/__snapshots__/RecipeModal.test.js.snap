// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal Component should render modal with data 1`] = `ShallowWrapper {}`;
